### ISSUE TITLE
Working solution for modern chart.js setup (React)

### ISSUE THREAD

**Issue by sarates** [AbelHeinsbroek/chartjs-plugin-crosshair#117]
Has anyone managed to find a working solution for synced crosshairs with "chart.js": "^4.1.1" and "react-chartjs-2": "^5.1.0" setup? Thanks!

**Comment by ViRuSTriNiTy** [comment]
I guess you just have to implement the crosshair by yourself:

https://www.youtube.com/watch?v=Ptg-4p5p_1Q

**Comment by nategrift** [comment]
A new version was created to support v4 of `chart.js`. I am not sure about `react-chartjs-2` but if it doesn't work with latest version of this plugin, please open another issue and we can resolve it there.

**Comment by joeuhren** [comment]
I'm also not sure about `react-chartjs-2` but just a note that the proposed fix by @nategrift above only works when the charts are the same height. If one chart is taller than another, the sync seems to work at first but can break while going back and forth across the data points so that the 2nd chart just stops moving in sync until you mouse out and back in to reset it.

The first and only repo I've seen that contains a fix for variable height charts is [https://github.com/TrevorLeeman/chartjs-plugin-crosshair/](https://github.com/TrevorLeeman/chartjs-plugin-crosshair/). The fix works great for the newest chartjs (4.4.7 at the time of writing) although it requires that you add an additional line to the chart config like this:

```
plugins: {
  crosshair: {
    sync: {
      axis: 'x', // Sync tooltips across charts of variable heights. Possible options: 'x' | 'xy'. Default: 'xy'
    }
  }
}
```

A big thank you to TrevorLeeman's and everyone else who originally helped with different variations on a fix for this bug!

For anyone looking for a quick way to take advantage of the fixed version in your own projects, I forked TrevorLeeman's repo and added the fixed js files to the project so that they can be used via cdn. Feel free to directly link to my repo via cdn link with [https://cdn.jsdelivr.net/gh/joeuhren/chartjs-plugin-crosshair@v2.0.5/build/chartjs-plugin-crosshair.min.js](https://cdn.jsdelivr.net/gh/joeuhren/chartjs-plugin-crosshair@v2.0.5/build/chartjs-plugin-crosshair.min.js). Don't forget to add the plugins > crosshair > sync > axis config line if you are going to be using variable chart heights.

Also note that I do not plan to maintain my fork of the crosshair plugin. If it happens to break again in a future chartjs version, I would recommend staying with whatever version of chartjs continues to work until another fix is found.
